src/platform/intel/cavs/platform.c

```diff
@@ -516,6 +519,12 @@ int platform_init(struct sof *sof)
 
 void platform_wait_for_interrupt(int level)
 {
-	/* TODO: go to LPS flow if pm-runtime is enabled for DSP */
+#if (CONFIG_CAVS_LPS)
+	if (pm_runtime_is_active(PM_RUNTIME_DSP, PLATFORM_MASTER_CORE_ID))
+		arch_wait_for_interrupt(level);
+	else
+		lp_wait_for_interrupt(level);
+#else
 	arch_wait_for_interrupt(level);
+#endif
```

Comment:
the code is somewhat odd, shouldn't the LPS part be in the architecture part?

Reply:
LPS is part of the cAVS platform, not the core xtensa architecture. Therefore I decided to put the code inside the platform. As I wrote in some of the previous comments, it would be nice to have a complete abstraction of the architecture in the lps code, however there is no other architecture integrated on cAVS platforms at the moment, so it could be deferred until/if other architectures are enabled imo.